### PR TITLE
Log message nan data in init

### PR DESCRIPTION
**Pull request by znicholls**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added (N/A)
- [x] Description in RELEASE_NOTES.md Added

## Adding to RELEASE_NOTES.md (remove section after adding to RELEASE_NOTES.md)

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

# Description of PR

A log message is now thrown if any null data is found on initialisation so that users know it's being dropped (I get caught out a lot being like, 'where has all my data gone'). Inspired by #291 and discussions in #286

### CODE REVIEW

**Comment by danielhuppmann** [comment]
Thanks @znicholls for this PR. I can't test it right now because I don't see any logging output (see #293), but I am worried about confusion this may create for non-expert users.

We are aiming `pyam` (also) at a not-so-sophisticated user base, and with the proposed change, one would see a warning message (about something being dropped) if you load a snapshot (wide format) with some variables reported at a yearly resolution (e.g., temperature output from a stylized climate model) and other variables reported  at decadal resolution (e.g., IAM output). This could easily be misunderstood as the snapshot being corrupted or incomplete.

**Comment by znicholls** [comment]
> This could easily be misunderstood as the snapshot being corrupted or incomplete.

That's fair (but one could argue the snapshot is somehow incomplete as not everything is on the same timestep). Happy to also close this/take another route (only throw the warning if the dataframe is actually empty once you remove nan rows?).

**Comment by danielhuppmann** [comment]
> Only throw the warning if the dataframe is actually empty once you remove nan rows.

Writing to log already happens if an `IamDataFrame` is empty after `filter()`, see [here](https://github.com/IAMconsortium/pyam/blob/103b60f8f2f46fddae4d0325cbc2651cb098030a/pyam/core.py#L1001) - I think it makes sense to replicate the same in `__init__()`.

Maybe it makes sense to add an `empty` attribute similar to [pandas.DataFrame.empty](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.empty.html)?

**Comment by znicholls** [comment]
> Writing to log already happens if an `IamDataFrame` is empty after `filter()`, see [here](https://github.com/IAMconsortium/pyam/blob/103b60f8f2f46fddae4d0325cbc2651cb098030a/pyam/core.py#L1001) - I think it makes sense to replicate the same in `__init__()`.

Nice will do that then

**Comment by znicholls** [comment]
> Maybe it makes sense to add an `empty` attribute similar to [pandas.DataFrame.empty](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.empty.html)?

Smart

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 
 # Next Release
 
+- [#290](https://github.com/IAMconsortium/pyam/pull/290) Add warning message if nan rows are detected in init data
```
Comment (danielhuppmann) [inline RELEASE_NOTES.md]:
```suggestion
- [#290](https://github.com/IAMconsortium/pyam/pull/290) Add warning message if `data` is empty at initialization (after formatting)
```

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 
 # Next Release
 
+- [#290](https://github.com/IAMconsortium/pyam/pull/290) Add warning message if `data` is empty at initialization (after formatting)
```
Comment (danielhuppmann) [inline RELEASE_NOTES.md]: just noticed one more little issue here...

```suggestion
- [#292](https://github.com/IAMconsortium/pyam/pull/292) Add warning message if `data` is empty at initialization (after formatting)
```

**Comment by danielhuppmann** [comment]
thanks @znicholls for this nice improvement - better user experience in edge-cases (or user errors...)

**Pull request merged**